policy_definitions/network_security/policy_definition_qby_allow_subnet_nsg.json

```
@@ -0,0 +1,287 @@
+{
+  "type": "Microsoft.Authorization/policyDefinitions",
+  "name": "QBY-Allow-Subnet-NSG",
+  "properties": {
+    "displayName": "Allow only Subnets with NSGs and correct names",
+    "description": "Checks if the subnet has the correct NSG attached to it. Test Subnet and NSG for Naming Convention.",
+    "policyType": "Custom",
+    "mode": "Indexed",
```

Subnets doesn't support tags
```suggestion
    "mode": "All",
```